app/models/host/base.rb

```diff
@@ -416,7 +416,7 @@ def get_interface_scope(name, attributes, base = self.interfaces)
 
           mac_based = base.where(:mac => macaddress)
           if attributes[:virtual]
-            mac_based.virtual.where(:identifier => name)
+            mac_based.virtual.where(:identifier => name) || base.where(:attached_to => mac_based.identifier)
```

Comment:
Isn't mac_based a collection? I think `.identifier` will fail

also wouldn't it override any interface attached to this identifier?